### ISSUE TITLE
Make helix msbuild tasks Cancelable

### ISSUE THREAD

**Issue by alexperovich** [dotnet/arcade#2130]
@alexperovich commented on [Tue Nov 27 2018](https://github.com/dotnet/core-eng/issues/4714)

All the tasks in the Helix SDK need to be cancellable to support canceling the build cleanly. https://stackoverflow.com/a/9967902 is how to do this.

**Comment by ericstj** [comment]
We should probably look at all tasks in Arcade that are potentially long running.

**Comment by safern** [comment]
This is how I did it for `DownloadResultsFromContainerTask` https://github.com/dotnet/arcade/pull/2103/commits/85c8b655761f349fe1914101e1e2a90a47d8d96c

That one is missing to pass the cancellation token down to `CloudBlob.DownloadToFileAsync` from Azure SDK. However to do that, we need to update dependencies and include new packages since the azure SDK split their packages into multiple, and also they depend on a higher version of Newtonsoft. This is what I tried doing, but it was failing when using the `AzureDevOpsTask` failing to load the new Newtonsoft assembly. 

https://github.com/dotnet/arcade/commit/49cb506ea8e8f27c73a8a70fccb01f1e5b30245c
